src/main/java/com/uber/profiling/ProfilerRunner.java

```diff
@@ -27,6 +28,7 @@
     
     private final Profiler profiler;
     private final AtomicLong errorCounter = new AtomicLong(0);
+    private ScheduledFuture<?> handler;
```

Comment:
This "handler" field is only used by setter/getter. Kind of weird. Any reason add it here?

Reply:
This is needed to allow the the profilers to be "rescheduled" in the Agent, that is - cancel the previous registration and start a new one if profiling frequency changes

Reply:
I do not see your code invoke setHandler/getHandler. Maybe I miss something?

Reply:
AgentImpl#scheduleProfilers calls both

Reply:
Yeah, sorry I missed that previously. It is still weird that we only have setHandler/getHandler in this class. Looks like we should move the "handler" out of this class.

Reply:
Ok - refactored this even more and actually get rid of the ProfilerRunner altogether.
Made the Profiler itself Runnable.